src/screens/RegistrationScreen/ClientRegistrationScreen.tsx

```diff
@@ -24,256 +25,172 @@ import {
-		if (!city) { Alert.alert('Please enter your city.'); return; }
-		if (!(/^\d{5}$/.test(zip))) { Alert.alert('Please enter a valid 5-digit zip code.'); return; }
-		if (!transportationMethod) { Alert.alert('Please select your preferred method of transportation.'); return; }
-		if (!termsOfService) { Alert.alert('Please read and accept the terms of service to complete your registration.'); return; }
+	// object contains the error status and corresponding message
+	// modifications in errorObj trigger change in UI.
+	const errorObj = {
+		email: {
+			isTrue: false,
+			message: 'Unknown error occurred',
+		},
+		password: {
+			isTrue: false,
+			message: 'Unknown error occurred',
+		},
+		retypedPassword: {
+			isTrue: false,
+			message: 'Unknown error occurred',
+		},
+		firstName: {
+			isTrue: false,
+			message: 'Unknown error occurred',
+		},
+		lastName: {
+			isTrue: false,
+			message: 'Unknown error occurred',
+		},
+	};
 
-		const statusCode = await register({
-			email, password, street, city, state, zip, transportationMethod, ethnicity, gender,
-		});
-		switch (statusCode) {
-			case (201 || 202): Alert.alert('Registration complete! Please log in to continue.'); navigate('LoginScreen', { email, password }); return;
-			case 406: Alert.alert('Error: not accepted'); return;
-			case 500: Alert.alert('Internal server error, please try again later.'); return;
-			default: Alert.alert("Sorry, that didn't work, please try again later."); console.log(statusCode);
-		}
+
+	const validateAndSubmit = async () => {
+		// outdated code, commented for future reference
+		// if (!email.includes('@') || !email.includes('.')) { Alert.alert('Please enter a valid email address.'); return; }
+		// if (password.length < 8) { Alert.alert('Please enter a password at least 8 characters long.'); return; }
+		// if (!street || street.split(' ').length < 3) { Alert.alert('Please enter your street number and name.'); return; }
+		// if (!city) { Alert.alert('Please enter your city.'); return; }
+		// if (!(/^\d{5}$/.test(zip))) { Alert.alert('Please enter a valid 5-digit zip code.'); return; }
+		// if (!transportationMethod) { Alert.alert('Please select your preferred method of transportation.'); return; }
+		// if (!termsOfService) { Alert.alert('Please read and accept the terms of service to complete your registration.'); return; }
+		//
+		// const statusCode = await register({
+		// email, password, street, city, state, zip, transportationMethod, ethnicity, gender,
+		// });
+		// switch (statusCode) {
+		// 	case (201 || 202): Alert.alert('Registration complete! Please log in to continue.'); navigate('LoginScreen', { email, password }); return;
+		// 	case 406: Alert.alert('Error: not accepted'); return;
+		// 	case 500: Alert.alert('Internal server error, please try again later.'); return;
+		// 	default: Alert.alert("Sorry, that didn't work, please try again later."); console.log(statusCode);
+		// }
```

Comment:
Are you leaving this commented because we'll be uncommenting it eventually?  Could we get the client side validation included as part of this commit?

Reply:
Just comment it otherwise lint would be angry, and those are outdated lines could be used for reference in data communication part we would implement later. It would be removed once we have registration runs well in the future. My intention was to get the UI side done in this PR and create another PR for communication. Since client validation is done by @emarti43 in other PRs, I think it's natural to create separate one for UI update. How do you think about it?